libs/networking/http_helpers/http_helpers.ml

```diff
@@ -0,0 +1,122 @@
+open Cohttp
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(* A few helpers to perform http GET and POST requests.
+ *
+ * Below we separate the methods out by async (returns Lwt promise),
+ * and sync (runs async method in lwt runtime)
+ * This way we can use the async methods in the language server,
+ * and other places too.
+ *
+ * Note that using [@@profiling] with xxx_async function is useless
+ * as the actual computation is done in the caller doing the
+ * Lwy_main.run
+ *)
+
+(*****************************************************************************)
+(* Client *)
+(*****************************************************************************)
+
+(* Create a client reference so we can swap it out with a testing version *)
+(*****************************************************************************)
+(* Async *)
+(*****************************************************************************)
+let client_ref : (module Cohttp_lwt.S.Client) option ref = ref None
+
+module Make (Lwt_wrapper : sig
+  val run : 'a Lwt.t -> 'a
+end) =
+struct
+  let get_async ?(headers = []) url =
+    Logs.debug (fun m -> m "GET on %s" (Uri.to_string url));
+    let module Client =
+      (val match !client_ref with
+           | Some client -> client
+           | None -> failwith "HTTP client not initialized")
```

Comment:
Potentially good to have a note here wrt virtual modules?

Reply:
I'll leave a comment explaining this